### PR TITLE
sched: Add k8s scheduler support

### PR DESCRIPTION
**Pull request by gaocegege**

### CODE REVIEW

File: treadmill/master.py [hunk 1]
```diff
@@ -99,9 +99,16 @@ def time_past(when):
 class Master(object):
     """Treadmill master scheduler."""
 
-    def __init__(self, zkclient, cellname, events_dir=None):
+    def __init__(self, zkclient, cellname,
+                 scheduler_vendor='native', config=None, events_dir=None):
+        if scheduler_vendor == 'k8s':
```
Comment (ebourgeois) [inline treadmill/master.py]: If we have multiple vendors, then this should be an Enum

Comment (ebourgeois) [inline treadmill/master.py]: Regardless, it shoudl be a global variable.

Comment (gaocegege) [inline treadmill/master.py]: Yeah, it is a good suggestion, I will update soon.

**Comment by gaocegege** [comment]
Hi, all.

I will update the code soon to fix:

* > we do not use relative imports (sometime we do use relative import as in "from . import x" but for "from ... import x" - rather from treadmill.utils import x.
* > we do not import classes, just packages.
* > If we have multiple vendors, then this should be an Enum.

And I will write the details about this PR in a few days and post in this issue :)

Thanks for your review.

**Comment by gaocegege** [comment]
Closing it since it is stale

**Pull request closed without merging**